### ISSUE TITLE
HTTP/1.1 503 Service Unavailable When Sending Test Email

### ISSUE THREAD

**Issue by Altoids94** [gophish/gophish#378]
I'm getting this issue when attempting to send out a test email:
![503](https://cloud.githubusercontent.com/assets/5906144/18766691/11f54ec4-80e1-11e6-8d17-44504aff3bed.PNG)

Here is the output from the terminal:
![503 2](https://cloud.githubusercontent.com/assets/5906144/18766751/56fedea4-80e1-11e6-8cfa-937cf0295e01.PNG)

Things were working fine when I was hosting my gophish server from internally my organization. However, after moving it to the DMZ, this issue appeared.

I tried Telnetting through port 25 (which is the port I'm using in my sending profile) to the Exchange server, and I'm able to send emails from the gophish server using Telnet without any problems. However, when I try to send the test email through the GoPhish UI, I get the 503 error.

Firewall rules are setup to allow port 443, 80, and 25.

I built from source and am using the latest edition.
Thanks!

**Comment by Altoids94** [comment]
Additional semi related question--
In config.json, why is there an option to change the SMTP settings if the Sending Profile designates which host we're using? What's the difference between the two?

**Comment by jordan-wright** [comment]
Disregard the SMTP settings in the config.json file.

I'm afraid I'm not sure what would be causing this to occur. For whatever reason, it appears the SMTP connection is receiving an invalid response and then closing the connection. Since gophish was working fine before moving it to the DMZ, it seems as though some network device in the path is messing with the connection.

**Comment by Altoids94** [comment]
Thanks, Jordan. I would think that too, but since I'm able to Telnet to the exchange server and send emails from the gophish server indicates that the connection should be working. It's only when I try from the application that I run into this error. 

Let me know if you can think of any other things that might be causing the problem.

**Comment by Altoids94** [comment]
So I figured out the issue and was able to resolve it. 

It turns out that the binary I was using at the time wasn't built from source (oops). I was using the pre-built binary from gophish_linux_64bit.zip. The issue was that for some reason, an extra ':25' was being added when it's sent to the mail relay, even if I omitted :25 from the sending profile. So whenever the mail server received the request, it would look like smtp.domain.com:25:25.

I didn't realize that the .zip files were so different from building from source. For example, in smtp.go of the models folder, the code is very different and a lot of revisions had been made. So lesson learned: Build from source!

Thanks for the great tool.
